src/zimdump.cpp

```diff
@@ -296,27 +319,51 @@ void ZimDumper::listArticleT(const zim::Article& article, bool extra)
   }
   std::cout << std::endl;
 }
+void write_to_error_directoy(const std::string& base, const std::string relpath, const char *content, size_t size)
+{
+    createdir(ERRORSDIR, base);
+    std::string url = relpath;
+
+    std::string::size_type p;
+    while ((p = url.find('/')) != std::string::npos)
+        url.replace(p, 1, "%2f");
+
+    std::ofstream stream(base + ERRORSDIR + url);
```

Comment:
`std::ofstream` doesn't handle non ascii path.
This is why we use the C `_wopen` in `write_to_file`. We must do the same in `write_to_error_directory`. (And even better, reuse the function `write_to_file` with the correct "error path" as argument).

Reply:
@mgautierfr we can use std::wofstream. It handles widechars.

Reply:
The problem is not the content of the file (we can use standard char) but the path of the file. And there is no constructor accepting a `wchar`/`wstring` for the filename.

Reply:
I see. But in this case, when we are creating a file we are not rendering text right? I think we should not care much about the encoding (UTF-8 on linux or UTF-16 on windows). Isn't windows taking a `char *` and do the internal encoding conversion?

Reply:
>  when we are creating a file we are not rendering text right?

I don't understand "rendering text".

>  Isn't windows taking a char * and do the internal encoding conversion?

No. And even if it was the case, the url is utf8, to we must correctly convert it.
The main idea in all libzim/kiwix project is to use utf8 everywhere. On windows, we must convert what we get from the system to utf8 just after "reading" it. On convert to widechars just before passing it to windows.

Reply:
By rendering I mean printing text on a screen. By doing that we can have more issues regarding locale. But yes, your right. If we used C++17 or C++20 we could avoid this issues :)

 [u8path](https://en.cppreference.com/w/cpp/filesystem/path/u8path)

Reply:
The problem is not about rendering, but interaction with outside (printing it or passing it to the system).

> If we used C++17 or C++20 we could avoid this issues :)

Yes, but we don't :)